externals/llvm-external-projects/torch-mlir-dialects/lib/Dialect/Tcp/IR/TcpOps.cpp

```diff
@@ -111,4 +111,52 @@ LogicalResult IsolatedGroupOp::verify() {
 
 OpFoldResult ConstOp::fold(FoldAdaptor) { return getValueAttr(); }
 
+LogicalResult ConcatOp::verify() {
+  auto outputTensorType = getResult().getType().cast<TensorType>();
+  int64_t axis = getAxis();
+  int64_t concatDimVal = outputTensorType.getShape()[axis];
+  // Accumulate concat dim, ignored if any of the inputs is dynamic.
+  int64_t concatDimAcc = 0;
+  // All tensors must have the smae dtype, rank and all non-concat dims must be
+  // the same.
+  for (auto type : getInputs().getTypes()) {
+    auto inputTensorType = type.cast<TensorType>();
+    if (inputTensorType.getElementType() != outputTensorType.getElementType())
+      return emitOpError() << "failed to verify tcp.concat with input and "
+                              "output operands rank mismatch";
+    if (outputTensorType.getRank() != inputTensorType.getRank())
+      return emitOpError() << "failed to verify tcp.concat with input and "
+                              "output operands rank mismatch";
```

Comment:
Can we replace this check with the `AllRanksMatch` trait? https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/IR/OpBase.td#L2451